### PR TITLE
Push fidp param set logic as high as possible in the sign in hook

### PR DESCRIPTION
**Pull request by pavinduLakshan**
### Purpose

$subject

### Related Issues
- N/A

### Related PRs
- N/A

### Checklist
- [ ] e2e cypress tests locally verified.
- [X] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [X] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [X] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/protected-app.tsx [hunk 1]
```diff
@@ -143,6 +144,28 @@ export const ProtectedApp: FunctionComponent<AppPropsInterface> = (): ReactEleme
...
 
+            const idToken: DecodedIDTokenPayload = await getDecodedIDToken();
+
+            const __experimental__platformIdP: {
+                enabled: boolean;
+                homeRealmId: string;
+            } = window["AppUtils"].getConfig()?.__experimental__platformIdP;
+
+            if (__experimental__platformIdP?.enabled) {
+                if (idToken?.default_tenant && idToken.default_tenant !== "carbon.super") {
```
Comment (Achintha444) [inline apps/console/src/protected-app.tsx:155]: if (__experimental__platformIdP?.enabled && idToken?.default_tenant && idToken.default_tenant !== "carbon.super") {

Comment (pavinduLakshan) [reply apps/console/src/protected-app.tsx]: Fixed with 5a6c282377

File: apps/console/src/protected-app.tsx [hunk 2]
```diff
@@ -143,6 +144,29 @@ export const ProtectedApp: FunctionComponent<AppPropsInterface> = (): ReactEleme
...
             let response: BasicUserInfo = null;
 
+            const idToken: DecodedIDTokenPayload = await getDecodedIDToken();
+
+            const __experimental__platformIdP: {
+                enabled: boolean;
+                homeRealmId: string;
+            } = window["AppUtils"].getConfig()?.__experimental__platformIdP;
+
+            if (__experimental__platformIdP?.enabled &&
```
Comment (pavinduLakshan) [inline apps/console/src/protected-app.tsx:154]: Check whether sub org console login is not affected by this.

File: apps/console/src/protected-app.tsx [hunk 3]
```diff
@@ -143,6 +144,29 @@ export const ProtectedApp: FunctionComponent<AppPropsInterface> = (): ReactEleme
...
+                idToken?.default_tenant &&
+                idToken.default_tenant !== "carbon.super"
```
Comment (pavinduLakshan) [inline apps/console/src/protected-app.tsx:156]: Use constant for "carbon.super"

**Review by Achintha444 — APPROVED** [review]

**Review by JayaShakthi97 — APPROVED** [review]

**Pull request merged**